From 2d2bdb7655e4aa085155bdef82fd70412ee94edb Mon Sep 17 00:00:00 2001
From: Axel Davy <axel.davy@ens.fr>
Date: Sat, 3 Jan 2015 21:41:04 +0100
Subject: [PATCH 19/21] d3dadapter: Add better check for DRI3

Try DRI3Open to see if DRI3 works.
Sometimes DRI3 is advertised, but isn't working.

Signed-off-by: Axel Davy <axel.davy@ens.fr>
---
 dlls/winex11.drv/dri3.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/dlls/winex11.drv/dri3.c b/dlls/winex11.drv/dri3.c
index e2f20b0..db41ac3 100644
--- a/dlls/winex11.drv/dri3.c
+++ b/dlls/winex11.drv/dri3.c
@@ -89,6 +89,7 @@ DRI3CheckExtension(Display *dpy, int major, int minor)
     xcb_dri3_query_version_reply_t *dri3_reply;
     xcb_generic_error_t *error;
     const xcb_query_extension_reply_t *extension;
+    int fd;
 
     xcb_prefetch_extension_data(xcb_connection, &xcb_dri3_id);
 
@@ -107,6 +108,12 @@ DRI3CheckExtension(Display *dpy, int major, int minor)
         return FALSE;
     }
 
+    if (!DRI3Open(dpy, DefaultScreen(dpy), &fd)) {
+        ERR("DRI3 advertised, but not working\n");
+        return FALSE;
+    }
+    close(fd);
+
     TRACE("DRI3 version %d,%d found. %d %d requested\n", major, minor, (int)dri3_reply->major_version, (int)dri3_reply->minor_version);
     free(dri3_reply);
 
-- 
2.2.2

